lib/pdf2img.js

```diff
@@ -79,43 +85,52 @@ Pdf2Img.prototype.convert = function(input, opts, callbackreturn) {
   }
 
   // Set output name
-  if (options.outputname) {
-    options.outputname = options.outputname;
-  } else {
+  if (!options.outputname) {
     options.outputname = output;
   }
 
   async.waterfall([
     // Get pages count
-        function (callback) {
-             gm(input).identify("%p ", function (err, value) {
-                var pageCount = String(value).split(' ');
-
-                if (!pageCount.length) {
-                    callback({
-                        result: 'error',
-                        message: 'Invalid page number.'
-                    }, null);
-                } else {
-                    // Convert selected page
-                    if (options.page !== null) {
-                        if (options.page <= pageCount.length) {
-                            callback(null, [options.page]);
-                        } else {
-                            callback({
-                                result: 'error',
-                                message: 'Invalid page number.'
-                            }, null);
-                        }
-                    } else  {
-                        callback(null, pageCount);
-                    }
-                }
-
-            })
-
-        },
-
+    function (callback) {
+      gm(input)
+        .options({timeout: options.timeoutMilliseconds})
+        .identify("%p ", function (err, value) {
+          if (err) {
+            if (err.message === 'gm() resulted in a timeout.') {
+              return callback({
+                result: 'timeout',
+                message: 'gm call took longer than ' + options.timeoutMilliseconds + 'ms.'
```

Comment:
Would be good to specify `gm identify` here and `gm convert` below, to indicate which step timed out.